first draft for RP2XXX docs

NitiKaur opened the pull request:
@mattytrentini @jimmo @dpgeorge   please review afd8ff5 . Also, I had this doubt regarding the troubleshooting section that I have mentioned on slack.

NitiKaur committed:
GPIO interrpt pins added for ESP32
Issue #6581

NitiKaur committed:
first draft for rp2xxx

mattytrentini reviewed (commented):
I've made a start on the review but will continue tomorrow - and we can discuss further at the meeting.

NitiKaur committed:
Updated genral.rst

NitiKaur committed:
Updated genral.rst

The suggested changes have been made along with some additional changes as discussed in the meeting( the reviewed and suggested changes made, language changed)

NitiKaur committed:
Updated quickref.rst

I have edited in places where changes were suggested plus the details have been cut short to include only the most necessary of information.

NitiKaur committed:
updated intro.rst

I have changes the language and wrote the matter in my own words

NitiKaur committed:
Update quickref.rst

NitiKaur commented:
@mattytrentini I have made the changes wherever it was suggested. @dpgeorge I have edited the quickref to include only the most relevant piece of information as well as written the three files in my own words as discussed over the meeting

mcauser commented on docs/rp2/general.rst:
```diff
@@ -10,29 +10,30 @@ the RP2040.
 -------------------------------------------
 
-If you want a deatiled technical specification, then you can view the `datasheets <https://datasheets.raspberrypi.org/rp2040/rp2040-datasheet.pdf>`_ 
+For a deatiled technical specification, then you can view the `datasheets <https://datasheets.raspberrypi.org/rp2040/rp2040-datasheet.pdf>`_ 
```
deatiled -> detailed

mcauser commented on docs/rp2/general.rst:
```diff
@@ -10,29 +10,30 @@ the RP2040.
+	* Each core has an integer divider peripheral and two interpolators
+# 264KB SRAM in six independent banks
+# No internal Flash or EEPROM memory (after reset, the bootploader loads firmware from either the external flash memory or USB bus into internal SRAM)
+# QSPI bus controller, which supports upto 16 MB of external Flash memory
```
upto -> up to

mcauser commented on docs/rp2/general.rst:
```diff
@@ -10,29 +10,30 @@ the RP2040.
-# QSPI bus controller, supporting up to 16 MB of external Flash memory
+	* Each core has an integer divider peripheral and two interpolators
+# 264KB SRAM in six independent banks
+# No internal Flash or EEPROM memory (after reset, the bootploader loads firmware from either the external flash memory or USB bus into internal SRAM)
```
bootploader?

mcauser commented on docs/esp32/quickref.rst:
```diff
@@ -603,3 +603,31 @@ the corresponding functions, or you can use the command-line client
+
+Interrupts
+----------
+**Hardware Interrupts** - They are fired by hardware peripherals and sources.hey can be: Internal or External. An example of internal hardware interrupts maybe something like a hardware timer interrupt or WDT. An example of an external hardware interrupt is external GPIO pins interrupts 
```
sources.hey -> sources. They

mcauser commented on docs/rp2/quickref.rst:
```diff
@@ -59,7 +59,16 @@ Use the :mod:`time <utime>` module::
 
-How do they work?
+The system timer peripheral on RP2040 provides a global microsecond timebase for the system, and generates
+interrupts based on this timebase. It supports • A single 64-bit counter, incrementing once per microsecond;• This counter can be read from a pair of latching registers, for race-free reads over a 32-bit bus;• Four alarms: match on the lower 32 bits of counter, IRQ on match.The timer uses a one microsecond reference that is generated in the watchdog and derived from the refernce clock,which itsel ...
```
A few `•` chars made it in there

mcauser commented on docs/rp2/quickref.rst:
```diff
@@ -92,11 +101,21 @@ See :ref:`machine.UART <machine.UART>`. ::
 
+.. note::
+ 	
+	REPL over UART is disabled by default.You can see the :ref:`rp2_intro` for details on how to enable REPL over UART.
```
default.You -> default. You

mcauser commented on docs/rp2/quickref.rst:
```diff
@@ -92,11 +101,21 @@ See :ref:`machine.UART <machine.UART>`. ::
+
+The channels can be clocked by external sources, using an edge-sensitive input mode for frequency measurement, and a level-sensitive input mode for duty cycle measurement.
+
+For more information on this topic you can see the PWM module in the rp2040 datasheet.
```
Add another link to it?
`datasheet <https://datasheets.raspberrypi.org/rp2040/rp2040-datasheet.pdf>`

mcauser commented on docs/rp2/quickref.rst:
```diff
@@ -112,7 +131,9 @@ Use the ``machine.PWM`` class::
-How does the ADC module work?
+The ADC on RP2040 has a 12-bits resolution but with MicroPython it is transformed to a 16-bit number.
+ 
+RP2040 has five ADC channels in total, four of which are brought out to chip GPIOs: GP26, GP27, GP28 and GP29. On Raspberry Pi Pico, the first three of these are brought out to GPIO pins. The fourth is used to measure the VSYS voltage on the board.The ADC’s fifth input channel is connected to a temperature sensor built into RP2040.
```
board.The -> board. The

mcauser commented on docs/rp2/quickref.rst:
```diff
@@ -208,7 +229,7 @@ See :ref:`machine.RTC <machine.RTC>` ::
 --------------------
 
-Is there a watchdog timer?
+The RP2XXX port even has a Watchdog.The watchdog is a countdown timer that can restart parts of the chip if it reaches zero. This can be used to restart the processor if software gets stuck in an infinite loop. The programmer must periodically write a value to the watchdog to stop it from reaching zero.
```
Watchdog.The -> Watchdog. The

mcauser commented on docs/rp2/tutorial/intro.rst:
```diff
@@ -1,6 +1,247 @@
+1.Getting started with MicroPython on the RP2xxx
+================================================
+
+The Pico board is the most popular in the series and we will be talking about it in general for all the RP2xxx boards.This tutorial will guide you through setting up MicroPython, connecting to the network and communicating with the Internet, using the hardware peripherals, and controlling some external components.
```
boards.This -> boards. This

mcauser commented on docs/rp2/tutorial/intro.rst:
```diff
@@ -1,6 +1,247 @@
+1.1. The MicroPython Environment
+---------------------------------
+
+Python is the fastest way to get started with embedded software on RP2xxx ports.Without compromising on approachability, MicroPython can be used for embedded systems development at a professional level.
```
ports.Without -> ports. Without

mcauser commented on docs/rp2/tutorial/intro.rst:
```diff
@@ -1,6 +1,247 @@
+---------------------------------
+
+Python is the fastest way to get started with embedded software on RP2xxx ports.Without compromising on approachability, MicroPython can be used for embedded systems development at a professional level.
+The arihmtic performance of the is quite accurate as the accelerated floating point libraries in RP2040’s on-board ROM storage are used automatically by the Python code.
```
arihmtic -> arithmetic

mcauser commented on docs/rp2/tutorial/intro.rst:
```diff
@@ -1,6 +1,247 @@
+Python is the fastest way to get started with embedded software on RP2xxx ports.Without compromising on approachability, MicroPython can be used for embedded systems development at a professional level.
+The arihmtic performance of the is quite accurate as the accelerated floating point libraries in RP2040’s on-board ROM storage are used automatically by the Python code.
+
+We also have ease of working on existing MicroPython projects as they can ported easily because of the standard *machine* module being used for most of the on-chip hardware.The second processor core is exposed through the *_thread* module.
```
hardware.The -> hardware. The

mcauser commented on docs/rp2/tutorial/intro.rst:
```diff
@@ -1,6 +1,247 @@
+1.4. Getting the firmware
+--------------------------
+
+To get MicroPython the easiest way, download the pre-built release binary from the `Pico Getting Started pages https://www.raspberrypi.org/documentation/rp2040/getting-started/`_. If you can’t or don’t want to use the pre-built release or you want to build your own MicroPython firmware binary like  developing your own C module for MicroPython — you can follow section 1.4 to  get the source code f ...
```
A few double-spaces

mcauser commented on docs/rp2/tutorial/intro.rst:
```diff
@@ -1,6 +1,247 @@
+1.5. Installing MicroPython on Raspberry Pi Pico
+-------------------------------------------------
+
+Raspberry Pi Pico has a BOOTSEL mode for programming firmware over the USB port.First step is to ensure that the board  is not plugged into any source of power: disconnect the micro USB cable if plugged in, and disconnect any other wires that might be providing power to the board, e.g. through the VSYS or VBUS pin. Now hold down the BOOTSEL button, and plug in the micro USB cable which must ne co ...
```
port.First -> port. First

mcauser commented on docs/rp2/tutorial/intro.rst:
```diff
@@ -1,6 +1,247 @@
+Raspberry Pi Pico has a BOOTSEL mode for programming firmware over the USB port.First step is to ensure that the board  is not plugged into any source of power: disconnect the micro USB cable if plugged in, and disconnect any other wires that might be providing power to the board, e.g. through the VSYS or VBUS pin. Now hold down the BOOTSEL button, and plug in the micro USB cable which must ne co ...
+
+Drag the MicroPython *firmware.uf2* file onto a drive called the RPI-RP2 which pops up. This programs the MicroPython firmware(the UF2 File) onto the flash memory on your Raspberry Pi Pico.
+After finishing,the board reboots and boots into MicroPython, waiting for further instructions.
```
finishing,the -> finishing, the

mcauser commented on docs/rp2/tutorial/intro.rst:
```diff
@@ -1,6 +1,247 @@
+
+.. note:: 
+
+	The following instructions assume that you are using a Raspberry Pi Pico. Some details may differ if you are building firmware for a different RP2040-based board. The board vendor should detail any extra 	    steps needed to build firmware for that particular board. These instructions are generic, but might differ at places like putting the default serial port on different pins,or including extr ...
```
```
extra 	    steps -> extra steps
```

mcauser commented on docs/rp2/tutorial/intro.rst:
```diff
@@ -1,6 +1,247 @@
+	$ cd portal/rp2
+	$ make
+
+After this,check for a directory called *build* ( *ports/rp2/build* relative to the micropython directory), which contains the new firmware binaries. The most important ones are:
```
this,check -> this, check

mcauser commented on docs/rp2/tutorial/intro.rst:
```diff
@@ -1,6 +1,247 @@
+
+.. note::
+
+	You can run ls /dev/tty* to list your serial ports. There may be quite a few, but MicroPython’s USB serial will start with /dev/ttyACM. If in doubt, unplug the micro USB connector and see which 		one disappears. If you don’t see anything, you can try rebooting your Raspberry Pi.
```
```
which 		one -> which one
```

mcauser commented on docs/rp2/tutorial/intro.rst:
```diff
@@ -1,6 +1,247 @@
+	Type "help()" for more information.
+	>>>
+
+This key combination tells MicroPython to reboot. You can do this at any time. When it reboots, MicroPython will printout a message saying exactly what firmware version it is running, and when it was built. Your version number will be different from the one shown here.
```
printout -> print out

mcauser commented on docs/rp2/tutorial/intro.rst:
```diff
@@ -1,6 +1,247 @@
+
+	REPL over UART is disabled by default.
+
+The MicroPython port for RP2040 does not expose REPL over a UART port by default. However this default can bechanged in the ports/rp2/mpconfigport.h source file.To use the REPL over UART first build MicroPython (see section 1.6 for details).
```
file.To -> file. To

mcauser commented on docs/rp2/tutorial/intro.rst:
```diff
@@ -1,6 +1,247 @@
+
+.. warning:: 
+
+	RX matches to TX, and TX matches to RX. You mustn’t connect the two opposite TX pins together, or the two RX pins. This is because MicroPython needs to listen on the channel that the Raspberry Pi 		transmits on, and vice versa.
```
```
Raspberry Pi 		transmits -> Raspberry Pi transmits
```

NitiKaur committed:
Final Update of general.rst

NitiKaur committed:
Final update quickref.rst

NitiKaur committed:
final update intro.rst

NitiKaur reviewed (commented):
@mcauser  I have made the changes too. I think it is all set for a merge

mcauser commented:
Hey @NitiKaur, looking good, only a few left:
docs/esp32/quickref.rst:609 `sources.hey` -> `sources. They`
docs/rp2/quickref.rst:224 `Watchdog.The` -> `Watchdog. The`
docs/rp2/tutorial/intro.rst:6 `boards.This` -> `boards. This`
docs/rp2/tutorial/intro.rst:105 `this,check` -> `this, check`

NitiKaur committed:
Update quickref.rst

NitiKaur committed:
Update quickref.rst

NitiKaur committed:
Update intro.rst

kadamski commented:
@NitiKaur It would make sense to state what you are doing in your commit messages. If you think about it, the "Update intro.rst" doesn't say much about the commit alone. For example, there are 5 different intro.rst files in the tree, so it would make sense to add "rp2: " prefix. Since it is documentation, the refix could be "docs/rp2: ". Then "update" is almost as generic as "change". Doesn't say what you change there. Maybe you fix something important, maybe you only fix some typos, or grammar or white spaces.

It also makes a lot of sense to group the changes logically - create a separate commit with simple fixes like whitespace fixes, typos, etc - changes that are obvious. Then have a separate commit with more serious changes that might be debatable or add completely new content and may require some deeper review.

Those are just advises, I'm not sure if this is required for this PR to be requested.

NitiKaur commented:
@kadamski Thank you for your suggestion. I will take care of it the nest time :)

NitiKaur commented:
Creating a new PR with the changes in ADC, PWM, Timer and watchdog class as discussed  in the new PR
